File: gen_deploy_multi.js
// gen_deploy_multi.js - Generate a single PHP script to deploy all files
const fs = require('fs');
const path = require('path');

const scriptDir = __dirname;

function getB64(file) {
    return fs.readFileSync(file).toString('base64');
}

const files = {
    'index.html': getB64(path.join(scriptDir, 'index.html')),
    'coupon.html': getB64(path.join(scriptDir, 'coupon.html')),
    'map.html': getB64(path.join(scriptDir, 'map.html')),
    'theme.css': getB64(path.join(scriptDir, 'theme.css')),
    'images/logo.png': getB64(path.join(scriptDir, 'images', 'logo.png')),
    'images/main_visual.jpg': getB64(path.join(scriptDir, 'images', 'main_visual.jpg'))
};

const php = `<?php
// deploy_multi.php - Multi-page Deployment Script
error_reporting(E_ALL);
ini_set('display_errors', 1);

$files = ${JSON.stringify(files, null, 4)};

foreach ($files as $name => $b64) {
    $path = __DIR__ . '/' . $name;
    $dir = dirname($path);
    if (!is_dir($dir)) {
        mkdir($dir, 0755, true);
    }
    file_put_contents($path, base64_decode($b64));
    echo "Deployed: $name<br>";
}

echo "<h1>Deployment Complete!</h1>";
echo "<a href='index.html'>Go to Site</a>";
?>`;

fs.writeFileSync(path.join(scriptDir, 'deploy_multi.php'), php, 'utf8');
console.log('deploy_multi.php generated.');

File: gen_deploy_multi.py
import base64
import os

def get_file_content(path):
    with open(path, 'r', encoding='utf-8') as f:
        return f.read()

def get_file_b64(path):
    with open(path, 'rb') as f:
        return base64.b64encode(f.read()).decode()

base_dir = r'c:\Users\NEC-PCuser\Documents\kazu_midnight_fortress\apollo-seikotsuin'
index_html = get_file_content(os.path.join(base_dir, 'index.html'))
coupon_html = get_file_content(os.path.join(base_dir, 'coupon.html'))
map_html = get_file_content(os.path.join(base_dir, 'map.html'))
theme_css = get_file_content(os.path.join(base_dir, 'theme.css'))
logo_b64 = get_file_b64(os.path.join(base_dir, 'images', 'logo.png'))
mv_b64 = get_file_b64(os.path.join(base_dir, 'images', 'main_visual.jpg'))

php_content = f"""<?php
// Apollo Seikotsuin - Multi-page Deployment Script
error_reporting(E_ALL);
ini_set('display_errors', 1);

echo "Starting deployment...<br>";

// Create images directory
if (!is_dir('images')) {{
    mkdir('images', 0755, true);
    echo "Created images directory.<br>";
}}

// Helper function to write files
function writeFile($path, $content) {{
    if (file_put_contents($path, $content) !== false) {{
        echo "Successfully wrote $path (" . strlen($content) . " bytes).<br>";
    }} else {{
        echo "Failed to write $path.<br>";
    }}
}}

// Write HTML/CSS files
writeFile('index.html', {repr(index_html)});
writeFile('coupon.html', {repr(coupon_html)});
writeFile('map.html', {repr(map_html)});
writeFile('theme.css', {repr(theme_css)});

// Write Binary files
writeFile('images/logo.png', base64_decode("{logo_b64}"));
writeFile('images/main_visual.jpg', base64_decode("{mv_b64}"));

echo "Deployment finished. <a href='index.html'>Go to Home</a>";
?>"""

with open(os.path.join(base_dir, 'deploy_multi.php'), 'w', encoding='utf-8') as f:
    f.write(php_content)

print("Generated deploy_multi.php")
